File: src-server/src/db/schemas/workspace.py
from typing import TYPE_CHECKING
from . import DTOBase

if TYPE_CHECKING:
    from .agent import AgentRead

class WorkspaceBase(DTOBase):
    name: str
    directory: str

class WorkspaceBrief(WorkspaceBase):
    id: int

class WorkspaceRead(WorkspaceBase):
    id: int
    workspace_background: str
    usable_agents: list[AgentRead]

class WorkspaceCreate(WorkspaceBase):
    workspace_background: str
    usable_agent_ids: list[int]

class WorkspaceUpdate(DTOBase):
    name: str | None
    directory: str | None
    workspace_background: str | None
    usable_agent_ids: list[int] | None

File: src-frontend/src/types/workspace.ts
import type { AgentRead } from "@/api/generated/schemas";

export type WorkspaceBase = {
  name: string;
  directory: string;
  workspace_background: string;
};

export type WorkspaceRead = WorkspaceBase & {
  id: number;
  usable_agents: AgentRead[];
};

export type WorkspaceCreate = WorkspaceBase & {
  usable_agent_ids: number[];
};

export type WorkspaceUpdate = Partial<
  WorkspaceBase & {
    usable_agent_ids: number[];
  }
>;
